### PR TITLE
Add tagged template literal methods

### LINKED ISSUE #4: Add template literal tags
Before:

```js
const g = require('escape-goat');
const escaped = `<html a=${g.escape(a)}><b>${g.escape(b)}</b> ${g.escape(c)}</html>`;
```

After:

```js
const g = require('escape-goat');
const escaped = g.escapeTag`<html a=${a}><b>${b}</b> ${c}</html>`;
```

The code required is minimal and this can be improved further:

```js
module.exports = function eskape(strs, ...vals) {
    var out = strs[0];

    for (let i = 0; i < vals.length; i++) {
        out = out + escape(vals[i]) + strs[i + 1];
    }

    return out;
}
```

> Code from https://github.com/wbinnssmith/eskape/blob/master/src.js

### PR DESCRIPTION
**Pull request by joelbandi**
Fixes #4

### CODE REVIEW

**Comment by sindresorhus** [comment]
@joelbandi ping :)

Also needs documentation.

**Comment by joelbandi** [comment]
@sindresorhus Sorry - Was on a break. I'll get to it right away

**Comment by joelbandi** [comment]
@sindresorhus I'll get to it right away

File: index.js [hunk 1]
```diff
@@ -13,3 +13,25 @@ exports.unescape = input => input
 	.replace(/&#39;/g, '\'')
 	.replace(/&lt;/g, '<')
 	.replace(/&gt;/g, '>');
+
+exports.escapeTag = function () {
+	const args = (arguments.length === 1 ? [arguments[0]] : Array.apply(null, arguments));
```
Comment (sindresorhus) [inline index.js]: Can be simplified:

```js
exports.escapeTag = function (input) {
	const values = [].slice.call(arguments, 1);
	let output = input[0];
	for (let i = 0; i < values.length; i++) {
		output = output + exports.escape(values[i]) + input[i + 1];
	}
	return output;
};
```

Same with the other method.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "escape-goat",
-  "version": "1.1.0",
+  "version": "1.2.0",
```
Comment (sindresorhus) [inline package.json]: Don't bump version in PRs. That's up to the maintainers to do.

File: readme.md [hunk 3]
```diff
@@ -40,6 +48,13 @@ Escapes the following characters in the given `input` string: `&` `<` `>` `"` `'
 
 Unescapes the following HTML entities in the given `input` string: `&amp;` `&lt;` `&gt;` `&quot;` `&#39;`
 
+### escapeGoat.escapeTag`template`
+
+escapes the following HTML characters in the given `input` template literal string: `&` `<` `>` `"` `'`
```
Comment (sindresorhus) [inline readme.md]: Escapes

**Comment by joelbandi** [comment]
@sindresorhus I made the required changes.

**Comment by sindresorhus** [comment]
Thanks :)

**Comment by joelbandi** [comment]
My first opensource PR merge! :smile:

**Pull request merged**